src/AWSSQS.jl

```diff
@@ -269,11 +269,21 @@ end
 ```
 """
 
+# sqs_messages(queue::AWSQueue) = AWSSQSMessages(queue)
+# Base.eltype(::Type{AWSSQSMessages}) = Dict{Symbol,Any}
+# Base.start(::AWSSQSMessages) = nothing
+# Base.done(::AWSSQSMessages, ::Any) = false
+# Base.next(q::AWSSQSMessages, ::Any) = (sqs_receive_message(q.queue), nothing)
+
 sqs_messages(queue::AWSQueue) = AWSSQSMessages(queue)
 Base.eltype(::Type{AWSSQSMessages}) = Dict{Symbol,Any}
 Base.start(::AWSSQSMessages) = nothing
-Base.done(::AWSSQSMessages, ::Any) = false
-Base.next(q::AWSSQSMessages, ::Any) = (sqs_receive_message(q.queue), nothing)
+Base.done(::AWSSQSMessages, state::Any) = state
+Base.next(q::AWSSQSMessages, ::Any) = try
+    (sqs_receive_message(q.queue), nothing)
+catch EOFError
+    (nothing,true)
+end
```

Comment:
I'm not sure about this.
Two things:
 1) With the recent move from `Requests.jl` to `HTTP.jl`, I don't think `EOFError` will be thrown anymore.
 2) I think of an SQS queue as being conceptually infinite. To me it seems cleaner for the interface to be an infinite iterator that can throw exceptions for exceptional situations, rather than having errors look like a "clean end".

Reply:
I didn't mean for that to be included in the PR - I agree with about the the SQS queue being conceptually infinite (although for my application it was more useful to be finite.)

Reply:
cool.
thx